scrabble.js

```diff
@@ -1,8 +1,203 @@
+POINTS = {
+      "A": 1,
+      "B": 3,
+      "C": 3,
+      "D": 2,
+      "E": 1,
+      "F": 4,
+      "G": 2,
+      "H": 4,
+      "I": 1,
+      "J": 8,
+      "K": 5,
+      "L": 1,
+      "M": 3,
+      "N": 1,
+      "O": 1,
+      "P": 3,
+      "Q": 10,
+      "R": 1,
+      "S": 1,
+      "T": 1,
+      "U": 1,
+      "V": 4,
+      "W": 4,
+      "X": 5,
+      "Y": 4,
+      "Z": 10
+    };
+
 var Scrabble = function() {};
 
-// YOUR CODE HERE
-Scrabble.prototype.helloWorld = function() {
-  return 'hello world!';
+var Score = function(){
+
+};
+
+Score.score = function(word){
+  var sum = 0;
+  var bonus = 0;
+  var splitToArray = word.toUpperCase().split("");
+  for(var i = 0; i < word.length; i++){
+    sum += (POINTS[splitToArray[i]]);
+  }
+  if (word.length == 7){
+    bonus = 50;
+  }
+  sum += bonus;
+  return sum;
+};
+
+Score.highest = function(arrayOfWords){
+
+  // This initializes this function so that there is a base to compare with.
+  var word = "";
+  var score = 0;
+
+  for(var i = 0; i < arrayOfWords.length; i++){
+
+    // Necessary to be able to use the .score. QUESTION - why is this necessary?
+    // var item = new Score(); when it is defined as a prototype
+
+    // Including word.length != 7 is key here because if word.length == 7 then the original word wins the tie breaker
+    if (Score.score(arrayOfWords[i]) == score && arrayOfWords[i].length < word.length && word.length !== 7){
+      word = arrayOfWords[i];
+      score = Score.score(arrayOfWords[i]);
+    } else if(Score.score(arrayOfWords[i]) > score){
+      word = arrayOfWords[i];
+      score = Score.score(arrayOfWords[i]);
+    } else if(Score.score(arrayOfWords[i]) == score && arrayOfWords[i].length == 7  && word.length !== 7){
+      word = arrayOfWords[i];
+      score = Score.score(arrayOfWords[i]);
```

Comment:
It seems like all pieces of this if statement are doing the same thing so is it really necessary?